arch/TriCore/TriCoreMapping.c

```diff
@@ -130,4 +121,120 @@ const char *TriCore_group_name(csh handle, unsigned int id)
 #endif
 }
 
+#ifndef CAPSTONE_DIET
+/// A LLVM<->CS Mapping entry of an operand.
+typedef struct insn_op {
+	uint8_t /* cs_op_type */ type; ///< Operand type (e.g.: reg, imm, mem)
+	uint8_t /* cs_ac_type */ access; ///< The access type (read, write)
+	uint8_t /* cs_data_type */
+		dtypes[10]; ///< List of op types. Terminated by CS_DATA_TYPE_LAST
+} insn_op;
+
+///< Operands of an instruction.
+typedef struct {
+	insn_op ops[16]; ///< NULL terminated array of operands.
+} insn_ops;
+
+const insn_ops insn_operands[] = {
+#include "TriCoreGenCSMappingInsnOp.inc"
+};
+
+static inline cs_op_type TriCore_get_op_type(MCInst *MI, unsigned OpNum)
+{
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+	assert(OpNum < ARR_SIZE(insn_operands[MI->Opcode].ops));
+
+	return insn_operands[MI->Opcode].ops[OpNum].type;
+}
+
+static inline cs_ac_type TriCore_op_access(MCInst *MI, unsigned OpNum)
+{
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+	assert(OpNum < ARR_SIZE(insn_operands[MI->Opcode].ops));
+
+	cs_ac_type access = insn_operands[MI->Opcode].ops[OpNum].access;
+	if (MCInst_opIsTied(MI, OpNum) || MCInst_opIsTying(MI, OpNum))
+		access |= (access == CS_AC_READ) ? CS_AC_WRITE : CS_AC_READ;
+	return access;
+}
 #endif
+
+void TriCore_set_access(MCInst *MI)
+{
+#ifndef CAPSTONE_DIET
+	if (!(MI->csh->detail == CS_OPT_ON && MI->flat_insn->detail))
+		return;
+
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+
+	cs_detail *detail = MI->flat_insn->detail;
+	cs_tricore *tc = &(detail->tricore);
+	for (int i = 0; i < tc->op_count; ++i) {
+		cs_ac_type ac = TriCore_op_access(MI, i);
+		cs_tricore_op *op = &tc->operands[i];
+		op->access = ac;
+		cs_op_type op_type = TriCore_get_op_type(MI, i);
+		if (op_type != CS_OP_REG) {
+			continue;
+		}
+		if (ac & CS_AC_READ) {
+			detail->regs_read[detail->regs_read_count++] = op->reg;
+		}
+		if (ac & CS_AC_WRITE) {
+			detail->regs_write[detail->regs_write_count++] =
+				op->reg;
+		}
+	}
+#endif
+}
+
+void TriCore_reg_access(const cs_insn *insn, cs_regs regs_read,
+			uint8_t *regs_read_count, cs_regs regs_write,
+			uint8_t *regs_write_count)
+{
+#ifndef CAPSTONE_DIET
+	uint8_t read_count, write_count;
+	cs_detail *detail = insn->detail;
+	read_count = detail->regs_read_count;
+	write_count = detail->regs_write_count;
+
+	// implicit registers
+	memcpy(regs_read, detail->regs_read,
```

Comment:
Could you add a `assert(sizeof(cs_regs) <= sizeof(insn->detail->regs_read));` here. Also below.
Just in case someone plays with the arrays sizes in the future.

Reply:
But this assertion is not True

```
/// Type of array to keep the list of registers
typedef uint16_t cs_regs[64];
```

```
uint16_t regs_read
	[MAX_IMPL_R_REGS=20]; ///< list of implicit registers read by this insn
```

Reply:
Sorry, was in a rush. Meant it the other way around.